cwms-data-api/src/main/java/cwms/cda/data/dao/RatingSetDao.java

```diff
@@ -83,11 +101,96 @@ private static String extractOfficeId(String ratingSet) throws JsonProcessingExc
         return office;
     }
 
+    public RatingSet retrieveLatest(RatingSet.DatabaseLoadMethod method, String officeId,
```

Comment:
Isn't this what `cwms_rating.retrieve_eff_ratings_xml` does? Can you verify with Mike Perryman?

Reply:
I'll double check with Perryman

Reply:
No, the Retrieve_Eff_Ratings_Xml lets you specify times that the ratings are in effect instead of the specific effictive dates for the time window. The fastest I've been able to retrieve the XML for the most recent rating for a rating specification is:
```
declare
    l_max_date  date;
    l_office_id cwms_v_rating.office_id%type := 'SWT';
    l_rating_id cwms_v_rating.rating_id%type := 'TULA.Stage;Flow.EXSA.PRODUCTION';
    l_xml       clob;
begin
    select max(effective_date)
      into l_max_date
      from cwms_v_rating
     where office_id = l_office_id
       and rating_id = l_rating_id;
    l_xml := cwms_rating.retrieve_ratings_xml_f(
        p_spec_id_mask         => l_rating_id,
        p_effective_date_start => l_max_date,
        p_effective_date_end   => l_max_date,
        p_time_zone            => 'UTC',
        p_office_id_mask       => l_office_id);
end;
/

```